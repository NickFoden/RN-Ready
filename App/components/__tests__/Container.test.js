import React from "react";
import { View } from "react-native";
import renderer from "react-test-renderer";

import { Container, styles } from "../Container";

it("exports a styles object", () => {
  expect(typeof styles).toBe("object");
});

it("renders succesfully without children", () => {
  const rendered = renderer.create(<Container />).toJSON();
  expect(rendered).toBeTruthy();
});

it("renders children props", () => {
  const rendered = renderer
    .create(
      <Container>
        <View />
      </Container>
    )
    .toJSON();
  expect(rendered).toMatchSnapshot();
});

it("uses the specified background color if provided", () => {
  const rendered = renderer
    .create(<Container backgroundColor="red" />)
    .toJSON();
  expect(rendered).toMatchSnapshot();
});
