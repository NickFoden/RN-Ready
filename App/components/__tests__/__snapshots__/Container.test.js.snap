// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children props 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#4F6D7A",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View />
</RCTSafeAreaView>
`;

exports[`uses the specified background color if provided 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#4F6D7A",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;
